pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4660,8 +4651,19 @@ cpdef to_offset(freq, bint is_period=False):
                         #  into the offset
                         offset *= stride_sign
                 else:
+                    if prefix in c_DEPR_ABBREVS:
+                        warnings.warn(
+                            f"\'{prefix}\' is deprecated and will be removed "
+                            f"in a future version. Please use "
+                            f"\'{c_DEPR_ABBREVS.get(prefix)}\' "
+                            f"instead of \'{prefix}\'.",
+                            FutureWarning,
+                            stacklevel=find_stack_level(),
+                        )
+                        prefix = c_DEPR_ABBREVS[prefix]
+
                     stride = int(stride)
-                    offset = _get_offset(name)
+                    offset = _get_offset(prefix)
```

Comment:
we had tests that didn't work for BH / CBH because we changed prefix to bh / cbh ` in line 4663 prefix = c_DEPR_ABBREVS[prefix]`, but then passed to `_get_offset` `name,` = `BH / CBH`. That is why I replaced name with prefix.